packages/edit-site/src/components/editor-canvas-container/index.js

```diff
@@ -66,6 +66,7 @@ function EditorCanvasContainer( { children, closeButtonLabel, onClose } ) {
 		if ( typeof onClose === 'function' ) {
 			onClose();
 		}
+		setCanvasMode( 'edit' );
```

Comment:
Switching to editor mode when the container is closed might not always be desirable. Separately to this PR, oven in https://github.com/WordPress/gutenberg/pull/50566, I'm exploring rendering the Style Book via the editor canvas container, but not within the `edit` mode. In that case, closing the Style Book shouldn't switch modes.

Is it possible to otherwise retain the behaviour in this PR without having to explicitly set the mode here?

Reply:
I agree with @andrewserong  - the editor canvas container component should be able to function in any editor "mode", and shouldn't therefore affect how other components use it. 

What about passing your desired functionality in the `onClose` prop?

Reply:
Thank you! sure, will update the PR.